include/pmacc/particles/operations/SetAttributeToDefault.hpp

```diff
@@ -38,15 +38,31 @@ namespace pmacc
 
         /** set an attribute to their default value
          *
-         * @tparam T_Partcile particle type
+         * @tparam T_DestParticleType particle type
+         *
+         * @{
          */
-        template<typename T_Worker, typename T_Particle>
-        HDINLINE void operator()(T_Worker const& worker, IdGenerator idGen, T_Particle& particle)
+
+        template<typename T_Worker, typename T_DestParticleType>
+        HDINLINE void operator()(T_Worker const& worker, IdGenerator idGen, T_DestParticleType& destParticle)
+        {
+            using ResolvedAttr = typename pmacc::traits::Resolve<Attribute>::type;
+            /* set attribute to it's user defined default value */
+            destParticle[Attribute()] = ResolvedAttr::getValue(worker, idGen, T_DestParticleType{});
+        }
+
+        template<typename T_Worker, typename T_DestParticleType, typename T_SrcParticleType>
+        HDINLINE void operator()(
```

Comment:
I think separating this to another functor, something like, `SetAttributeFromParticle` is maybe better than having an overload of `SetAttributeToDefault` as I think the current name is doesn't make sense.

Reply:
For short I thought about following your suggestion but I decided against it.

- Both methods are equivalent but for one no extern particle is required. Nevertheless the particle attributes were set to default.
- There is not need to encode the parameters into the name where `SetAttributeFromParticle` is exactly doing it.

I added a sentence to the doc string to point to value_identifier behaviors.

```
Check value_identifier and value_identifier_func for special behaviours on how defaults get derived.
```

Reply:
Fair. I am okay with keeping it like this.
But maybe the best would be to rename the functor `SetAttribute` with these overloads. 
Attribute is set from a particle and if no source particle is provided, it uses the defaults.

Reply:
> Attribute is set from a particle and if no source particle is provided, it uses the defaults.

Ohh I think there is some misunderstanding about what this functor is doing. It will always set all attributes to the default.
A value_identifier also always gets a source particle that can be taken into account or not to derive the default value.
`value_identifier` has always a constant default value. `value_identifier_func` can derive the default from the id generator and or the source particle, where it could be that the source particle is invalid at the moment while `SetAttributeToDefault` is executed.
In any case what we set is the default.